python/cudf/cudf/io/parquet.py

```diff
@@ -672,6 +697,12 @@ def _generate_filename():
     return uuid4().hex + ".parquet"
 
 
+def get_estimated_file_size(df):
+    df_mem_usage = df.memory_usage().sum()
+    file_size = df_mem_usage // 2.5
```

Comment:
was just curious where does this heuristic come from? Is it based off a known compression ratio?

Reply:
So, this was based on experiments on a data frame with no nulls and all unique values for almost all datatypes, the size of file will translate to 2.5x times the GPU memory usage once loaded into the GPU. Obviously, this is not a final heuristic that will work for all types of datasets, but we can improve it in the future or give an undocumented parameter so that customers can tweak it according to their needs if they see it under-cut a lot.